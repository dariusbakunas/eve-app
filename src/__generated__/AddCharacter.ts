/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { CharacterInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: AddCharacter
// ====================================================

export interface AddCharacter_addCharacter {
  __typename: "Character";
  id: string;
  name: string;
  scopes: string[] | null;
}

export interface AddCharacter {
  addCharacter: AddCharacter_addCharacter;
}

export interface AddCharacterVariables {
  input: CharacterInput;
}
