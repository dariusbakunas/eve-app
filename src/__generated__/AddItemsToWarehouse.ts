/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { WarehouseItemInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: AddItemsToWarehouse
// ====================================================

export interface AddItemsToWarehouse_addItemsToWarehouse_item {
  __typename: "InvItem";
  id: string;
}

export interface AddItemsToWarehouse_addItemsToWarehouse {
  __typename: "WarehouseItem";
  item: AddItemsToWarehouse_addItemsToWarehouse_item;
  quantity: number;
  unitCost: number;
}

export interface AddItemsToWarehouse {
  addItemsToWarehouse: AddItemsToWarehouse_addItemsToWarehouse[];
}

export interface AddItemsToWarehouseVariables {
  id: string;
  input: WarehouseItemInput[];
}
