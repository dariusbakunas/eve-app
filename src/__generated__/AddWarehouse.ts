/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: AddWarehouse
// ====================================================

export interface AddWarehouse_addWarehouse {
  __typename: "Warehouse";
  id: string;
  name: string;
}

export interface AddWarehouse {
  addWarehouse: AddWarehouse_addWarehouse;
}

export interface AddWarehouseVariables {
  name: string;
}
