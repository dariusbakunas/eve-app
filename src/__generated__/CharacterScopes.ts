/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: CharacterScopes
// ====================================================

export interface CharacterScopes_scopes {
  __typename: "Scope";
  id: string;
  name: string;
}

export interface CharacterScopes {
  scopes: CharacterScopes_scopes[];
}
