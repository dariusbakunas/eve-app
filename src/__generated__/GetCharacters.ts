/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: GetCharacters
// ====================================================

export interface GetCharacters_characters {
  __typename: "Character";
  id: string;
  name: string;
  scopes: string[] | null;
}

export interface GetCharacters {
  characters: GetCharacters_characters[];
}
