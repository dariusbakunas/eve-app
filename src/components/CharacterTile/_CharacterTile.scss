.character-tile-component {
  .character-tile-header {
    display: flex;
    align-items: center;

    .character-tile-header-avatar {
      flex: 0 0 auto;
      margin-right: $spacing-04;
    }

    .character-tile-header-content {
      flex: 1 1 auto;
      line-height: 1.25rem;
      span {
        display: block;
        &.dob {
          font-size: small;
        }
      }
    }
  }
}