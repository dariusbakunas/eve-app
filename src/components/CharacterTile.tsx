import React from 'react';
import { CardContent, makeStyles, Theme } from '@material-ui/core';
import CardHeader from '@material-ui/core/CardHeader';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import moment from 'moment';
import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import { GetCharacters_characters as Character } from '../__generated__/GetCharacters';

const useStyles = makeStyles<Theme>(theme => ({
  card: {
    height: 'calc(100% - 5px)',
    width: 'calc(100% - 5px)',
    display: 'flex',
    flexDirection: 'column',
  },
}));

interface CharacterTileProps {
  character: Character;
  onRemove: (id: string, name: string) => void;
  onUpdate: (character: Character) => void;
}

const CharacterTile: React.FC<CharacterTileProps> = ({ character, onRemove, onUpdate }) => {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const menuOpen = Boolean(anchorEl);
  const classes = useStyles();
  const { id, name, birthday, corporation, securityStatus, totalSp } = character;

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const handleRemove = () => {
    setAnchorEl(null);
    onRemove(id, name);
  };

  const handleUpdate = () => {
    setAnchorEl(null);
    onUpdate(character);
  };

  return (
    <React.Fragment>
      <Card className={classes.card}>
        <CardHeader
          avatar={
            <Avatar aria-label="recipe" className={classes.avatar} src={`https://image.eveonline.com/Character/${id}_128.jpg`}>
              {name[0]}
            </Avatar>
          }
          action={
            <IconButton aria-label="settings" onClick={handleClick}>
              <MoreVertIcon />
            </IconButton>
          }
          title={name}
          subheader={`Born: ${moment(birthday).format('YYYY-MM-DD LT')}`}
        />
        <CardContent>
          {corporation.alliance && (
            <div>
              <strong>Alliance:</strong> {corporation.alliance.name} [{corporation.alliance.ticker}]
            </div>
          )}
          <div>
            <strong>Corporation:</strong> {corporation.name} [{corporation.ticker}]
          </div>
          <div>
            <strong>Security status:</strong> {securityStatus.toFixed(2)}
          </div>
          {totalSp && (
            <div>
              <strong>Total SP:</strong> {totalSp.toLocaleString()}
            </div>
          )}
        </CardContent>
      </Card>
      <Menu
        id={`character-menu-${id}`}
        anchorEl={anchorEl}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        getContentAnchorEl={null}
        open={menuOpen}
        onClose={handleClose}
      >
        <MenuItem onClick={handleUpdate}>
          <Typography>Update Scopes</Typography>
        </MenuItem>
        <MenuItem onClick={handleRemove}>
          <Typography color="error">Remove</Typography>
        </MenuItem>
      </Menu>
    </React.Fragment>
  );
};

export default CharacterTile;
