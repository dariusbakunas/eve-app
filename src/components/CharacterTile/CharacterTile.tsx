import { Avatar } from "../Avatar/Avatar";
import { GetCharacters_characters as Character } from "../../__generated__/GetCharacters";
import { Tile } from "carbon-components-react";
import moment from "moment";
import React from "react";

interface ICharacterTile {
  character: Character;
}

export const CharacterTile: React.FC<ICharacterTile> = ({ character }) => {
  const { id, name, birthday, corporation, totalSp, securityStatus } = character;

  return (
    <Tile className="character-tile-component">
      <div className="character-tile-header">
        <div className="character-tile-header-avatar">
          <Avatar alt={name} src={`https://image.eveonline.com/Character/${id}_128.jpg`} />
        </div>
        <div className="character-tile-header-content">
          <h5>{name}</h5>
          <span className="dob">{`Born: ${moment(birthday).format("YYYY-MM-DD LT")}`}</span>
        </div>
        <div className="character-tile-header-action"></div>
      </div>
      <div className="character-tile-content">
        {corporation.alliance && (
          <div>
            <strong>Alliance:</strong> {corporation.alliance.name} [{corporation.alliance.ticker}]
          </div>
        )}
        <div>
          <strong>Corporation:</strong> {corporation.name} [{corporation.ticker}]
        </div>
        <div>
          <strong>Security status:</strong> {securityStatus.toFixed(2)}
        </div>
        {totalSp && (
          <div>
            <strong>Total SP:</strong> {totalSp.toLocaleString()}
          </div>
        )}
      </div>
    </Tile>
  );
};
