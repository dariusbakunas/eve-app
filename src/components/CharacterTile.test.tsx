import { GetCharacters_characters as Character } from '../__generated__/GetCharacters';
import { shallow } from 'enzyme';
import CharacterTile from './CharacterTile';
import React from 'react';

describe('CharacterTile', () => {
  it('match snapshot', () => {
    const character: Character = {
      __typename: 'Character',
      id: '1704528d-6ef0-4590-9108-02a900cfd075_128',
      birthday: '2008-09-14T17:19:00.000Z',
      corporation: {
        __typename: 'Corporation',
        id: '1704528d-6ef0-4590-9108-02a900cfd075_127',
        name: 'Krajcik - Conroy ',
        ticker: 'KC',
        alliance: {
          __typename: 'Alliance',
          id: '1704528d-6ef0-4590-9108-02a900cfd075_126',
          name: 'Bauch, Rippin and Lehner [Group]',
          ticker: 'BRL',
        },
      },
      name: 'Bartholome',
      scopes: ['esi-assets.read_assets.v1', 'esi-characterstats.read.v1'],
      securityStatus: -1.2,
      totalSp: 12231323,
    };
    const wrapper = shallow(<CharacterTile character={character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
