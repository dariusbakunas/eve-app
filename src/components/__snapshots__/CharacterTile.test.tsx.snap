// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterTile match snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-card-1"
  >
    <WithStyles(ForwardRef(CardHeader))
      action={
        <ForwardRef(WithStyles)
          aria-label="settings"
          onClick={[Function]}
        >
          <UNDEFINED />
        </ForwardRef(WithStyles)>
      }
      avatar={
        <ForwardRef(WithStyles)
          aria-label="recipe"
          src="https://image.eveonline.com/Character/1704528d-6ef0-4590-9108-02a900cfd075_128_128.jpg"
        >
          B
        </ForwardRef(WithStyles)>
      }
      subheader="Born: 2008-09-14 5:19 PM"
      title="Bartholome"
    />
    <WithStyles(ForwardRef(CardContent))>
      <div>
        <strong>
          Alliance:
        </strong>
         
        Bauch, Rippin and Lehner [Group]
         [
        BRL
        ]
      </div>
      <div>
        <strong>
          Corporation:
        </strong>
         
        Krajcik - Conroy 
         [
        KC
        ]
      </div>
      <div>
        <strong>
          Security status:
        </strong>
         
        -1.20
      </div>
      <div>
        <strong>
          Total SP:
        </strong>
         
        12,231,323
      </div>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "bottom",
      }
    }
    getContentAnchorEl={null}
    id="character-menu-1704528d-6ef0-4590-9108-02a900cfd075_128"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(Typography))>
        Update Scopes
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(Typography))
        color="error"
      >
        Remove
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</Fragment>
`;
